### PR TITLE
[TR] Merge running js specs and analysis using grunt into master

### PR DESCRIPTION
**Pull request by filipsalpe**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,10 @@ before_script:
     - sh -c 'if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then echo "extension = mongo.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini; fi;'
     - sh -c 'if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then composer require --no-update doctrine/mongodb-odm v1.0.0-beta12@dev; fi;'
     - sh -c 'if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then composer require --no-update doctrine/mongodb-odm-bundle v3.0.0-BETA6@dev; fi;'
-    - composer update --prefer-dist --no-scripts --no-interaction
+    - sh -c 'if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then composer update --prefer-dist --no-interaction; else composer update --prefer-dist --no-interaction --no-scripts; fi;'
+    - sh -c 'if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then ./app/console oro:requirejs:generate-config; fi;'
+    - npm install -g grunt-cli
+    - npm install
```
Comment (jmleroux) [inline .travis.yml]: Not needed for hhvm

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/jstree/jquery.jstree.tree_selector.js [hunk 2]
```diff
@@ -65,22 +67,22 @@
...
-            no_tree_message : false,
-            node_label_field : 'code',
-            preselect_node_id : false
+        defaults: {
+            ajax: false,
+            data: false,
+            tree_selector_buttons: false,
+            no_tree_message: false,
+            node_label_field: 'code',
+            preselect_node_id: false
```
Comment (jmleroux) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/jstree/jquery.jstree.tree_selector.js]: uneeded blank line

File: .jscsrc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    "disallowMultipleLineStrings": true,
+    "disallowMultipleVarDecl": true,
+    "requireSpaceBeforeBlockStatements": true,
+    "requireParenthesesAroundIIFE": true,
+    "requireSpacesInConditionalExpression": true,
+    "requireBlocksOnNewline": true,
+    "requireCommaBeforeLineBreak": true,
+    "requireSpaceBeforeBinaryOperators": true,
+    "requireSpaceAfterBinaryOperators": true,
+    // "requireCamelCaseOrUpperCaseIdentifiers": true,
```
Comment (willy-ahva) [inline .jscsrc]: normal it's commented ?

**Comment by jmleroux** [comment]
Could we not use composer to install jasmine ?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "pim-community-dev",
+  "version": "1.4.0",
```
Comment (nidup) [inline package.json]: "version": "1.4.x-dev",

File: .travis.yml (hunk 1 above)
Comment (filipsalpe) [inline .travis.yml]: needed: `if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then grunt travis; else grunt codestyle`

File: .jscsrc (hunk 3 above)
Comment (filipsalpe) [inline .jscsrc]: yeah, it's an option we'd likely use in the future, just too many errors to fix right now

File: package.json (hunk 4 above)
Comment (filipsalpe) [inline package.json]: not supported by npm - they have some validation algorithm on this

**Comment by filipsalpe** [comment]
@jmleroux dropped jasmine form src, npm takes care of installing it anyway

**Comment by jmleroux** [comment]
GTM

File: package.json (hunk 4 above)
Comment (nidup) [inline package.json]: ok

**Comment by nidup** [comment]
GTM

**Pull request merged**